### PR TITLE
harden assembly resolve during deserialization

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

https://jira.autodesk.com/browse/DYN-2160

This PR hardens graph deserialization when searching for a node type/assembly. Previously assembly load could throw an exception, we would not handle it and a dummy node would not be created.

Now we catch the exception, log it to Dynamo console, and in some cases even recover and create the node correctly if the assembly can be found by the `nodeFactory` when it calls `ResolveType` later in the `nodeReader` 

The main cause of this failure is when some other code attaches an AssemblyResolve handler and then throws from inside that handler - these handlers should not throw - they should either return null or return the assembly, but we cannot control other handlers in any host so we should harden the loading code.

 

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@ColinDayOrg 
@reddyashish

### CODE REVIEW

File: test/DynamoCoreTests/DummyNodeTests.cs [hunk 1]
```diff
@@ -91,6 +91,33 @@ public void WorkspaceIsReadonly_IfXmlDummyNodePresent()
...
+                if (i == -1)
+                {
+                    exceptionCount = exceptionCount + 1;
+                    throw new Exception("TestingTesting");
+                }
+                return null;
+            });
+            //attach our handler that will throw
+            AppDomain.CurrentDomain.AssemblyResolve += handler;
+            System.Threading.Thread.Sleep(5000);
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/DummyNodeTests.cs]: this is not needed, will remove.

**Comment by ColinDayOrg** [comment]
This PR LGTM (pending the checks above that have not passed yet).

**Comment by mjkkirschner** [comment]
just waiting on the self serve.

File: src/DynamoCore/Graph/Workspaces/SerializationConverters.cs [hunk 2]
```diff
@@ -97,7 +97,15 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
...
-            var type = Type.GetType(obj["$type"].Value<string>());
+            Type type = null;
+            try
+            {
+               type = Type.GetType(obj["$type"].Value<string>());
+            }
+            catch(Exception e)
+            {
+                nodeFactory?.AsLogger().Log(e);
+            }
```
Comment (angelowang) [inline src/DynamoCore/Graph/Workspaces/SerializationConverters.cs]: @mjkkirschner do we want to continue to look into the preloaded nodes for the types if the exception happens?

**Pull request merged**